docs/testing-operators.md

```diff
@@ -1,120 +1,414 @@
+```
+
+### 5. View Available Operators
+
+Once the `OperatorSource` is deployed, the following command can be used to list the available operators (until an operator is pushed into quay, this list will be empty):
+
+> The command below assumes `johndoe-operators` as the name of the `OperatorSource` object. Adjust accordingly.
+
+```
+kubectl get opsrc johndoe-operators -o=custom-columns=NAME:.metadata.name,PACKAGES:.status.packages -n marketplace
+
+NAME                PACKAGES
+johndoe-operators   my-operator
+```
+
+### 6. Create CatalogSourceConfig
+
+Once the OperatorSource has been added, a `CatalogSourceConfig` needs to be created in the `marketplace` namespace to make those Operators available on cluster.
+
+Create the following file as `catalog-source-config.yaml`:
+
+```
+apiVersion: operators.coreos.com/v1
+kind: CatalogSourceConfig
+metadata:
+  name: johndoe-operators
+  namespace: marketplace
+spec:
+  targetNamespace: olm
+  packages: my-operator
+```
+
+In the above example:
+
+* `olm` is a namespace that OLM is watching fot `CatalogSource` objects
+* `packages` is a comma-separated list of operators that have been pushed to quay.io and should be deployable by this source.
+
+> The file above assumes `my-operators` as the name of the operator bundle. Adjust accordingly.
+
+Deploy the `CatalogSourceConfig` resource:
+
+```
+kubectl apply -f catalog-source-config.yaml
+```
+
+When this file is deployed, a `CatalogSourceConfig` resource is created in the `marketplace` namespace.
+
+```
+kubectl get catalogsourceconfig -n marketplace
+
+NAME                      STATUS      MESSAGE                                       AGE
+johndoe-operators         Succeeded   The object has been successfully reconciled   93s
+```
+
+Additionally, a `CatalogSource` is created in the namespace indicated in `spec.targetNamespace` (in the above example, `olm`):
+
+```
+kubectl get catalogsource -n olm
+
+NAME                           NAME        TYPE   PUBLISHER   AGE
+johndoe-operators              Custom      grpc   Custom      3m32s
+[...]
+```
+### 7. Create an OperatorGroup
+
+An `OperatorGroup` is used to denote which namespaces your Operator should be watching. It must in the namespace where your operator should be deployed, we'll use `default` in this example.
+
+Its configuration depends on your Operator supporting watching its own namespace, a single namespace or all namespaces (as indicated by `spec.installModes` in the CSV).
+
+Create the following file as  `operator-group.yaml` if your Operator supports watching its own or a single namespace.
+
+If your Operator supports watching all namespaces you can omit the following step and place your `Subscription` (see next step) in the `operators` namespace instead.
+
+```
+apiVersion: operators.coreos.com/v1alpha2
+kind: OperatorGroup
+metadata:
+  name: my-operatorgroup
+  namespace: default
```

Comment:
Question: We can't have TooManyOperatorGroups under the same namespace.https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/operatorgroups.md#toomanyoperatorgroups
In that case, do we check if there is already an OperatorGroup under default NS and append new NS under target namespace:
  -default
  - foo
or  do we create new operatorgroup  with namespace and targetnamespace  as foo?